lib/internal/bootstrap/loaders.js

```diff
@@ -208,6 +210,81 @@
         NativeModule.require;
       fn(this.exports, requireFn, this, process);
 
+      if (config.experimentalModules) {
+        this.namespace = Object.keys(this.exports);
+
+        const update = (property, value) => {
+          if (this.reflect !== undefined && this.namespace.includes(property))
+            this.reflect.exports[property].set(value);
+        };
+
+        const methodWrapMap = new WeakMap();
+
+        const wrap = (target, name, value) => {
+          if (typeof value !== 'function')
+            return value;
+
+          if (typeof value.name === 'string' && /^bound /.test(value.name))
+            return value;
+
+          if (methodWrapMap.has(value))
+            return methodWrapMap.get(value);
+
+          const p = new Proxy(value, {
+            apply(t, thisArg, args) {
+              if (thisArg === proxy)
+                thisArg = target;
+              return Reflect.apply(t, thisArg, args);
```

Comment:
You might end up needing a `thisArg === nsObj` check.
A test for calling a native method on the namespace object would cover it.